state/state.go

```diff
@@ -762,3 +762,21 @@ func (st *state) AllPendingTxs() []*tx.Tx {
 
 	return st.txPool.AllPendingTxs()
 }
+
+func (st *state) IsPruned() bool {
+	st.lk.RLock()
+	defer st.lk.RUnlock()
```

Comment:
`IsPruned` is only set when the node is running and does not have any writes during application execution. Therefore, there is no need to lock it here.